nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/UnpackContent.java

```diff
@@ -473,24 +495,28 @@ protected void processEntry(final InputStream zipInputStream, final boolean dire
         private static class CompressedZipInputStreamCallback extends ZipInputStreamCallback {
 
             private final boolean allowStoredEntriesWithDataDescriptor;
+            private final Charset filenameEncoding;
 
             private CompressedZipInputStreamCallback(
                     final Pattern fileFilter,
                     final ProcessSession session,
                     final FlowFile sourceFlowFile,
                     final List<FlowFile> unpacked,
                     final String fragmentId,
-                    final boolean allowStoredEntriesWithDataDescriptor
+                    final boolean allowStoredEntriesWithDataDescriptor,
+                    final Charset filenameEncoding
             ) {
                 super(fileFilter, session, sourceFlowFile, unpacked, fragmentId);
                 this.allowStoredEntriesWithDataDescriptor = allowStoredEntriesWithDataDescriptor;
+                this.filenameEncoding = filenameEncoding;
             }
 
             @Override
             public void process(final InputStream inputStream) throws IOException {
-                try (final ZipArchiveInputStream zipInputStream = new ZipArchiveInputStream(new BufferedInputStream(inputStream), null, true, allowStoredEntriesWithDataDescriptor)) {
+                try (final ZipArchiveInputStream zipInputStream = new ZipArchiveInputStream(new BufferedInputStream(inputStream),
+                    filenameEncoding.toString(), true, allowStoredEntriesWithDataDescriptor)) {
                     ZipArchiveEntry zipEntry;
-                    while ((zipEntry = zipInputStream.getNextZipEntry()) != null) {
+                    while ((zipEntry = zipInputStream.getNextEntry()) != null) {
```

Comment:
Same ask as in the tar case.  Why change the method call here?

Reply:
I will revert these method name changes